packages/fxa-settings/src/components/ConnectedServices/index.tsx

```diff
@@ -55,9 +73,156 @@ export const ConnectedServices = () => {
     setErrorText('Sorry, there was a problem refreshing the recovery key.');
     alertBar.show();
   });
-
   const showMobilePromo = !sortedAndUniqueClients.filter(isMobileDevice).length;
 
+  // TODO: should this view name be the exact same as in old-settings? "settings.clients.disconnect"?
+  const viewName = 'settings.clients.disconnect';
+
+  // The Confirm Disconnect modal is shown when a user clicks 'Sign Out' on a sync service.
+  // It asks the user to confirm they want to disconnect, and answer a survey question explaining
+  // why they are disconnecting.
+  const [
+    confirmDisconnectModalRevealed,
+    revealConfirmDisconnectModal,
+    hideConfirmDisconnectModal,
+  ] = useBooleanState();
+  const [submitStartTime, setSubmitStartTime] = useState<number>();
+
+  // After the user confirms they want to disconnect from sync in Confirm Disconnect modal,
+  // if their reason was a lost/stolen device, or a suspicious device, then we show them
+  // an informative modal with some advice on next steps to take.
+  const [
+    adviceModalRevealed,
+    revealAdviceModal,
+    hideAdviceModal,
+  ] = useBooleanState();
+
+  const clearDisconnectingState = (
+    errorMessage?: string,
+    error?: ApolloError
+  ) => {
+    hideConfirmDisconnectModal();
+    DS.client = null;
+    DS.reason = null;
+    setSubmitStartTime(0);
+    if (errorMessage) {
+      alertBar.error(errorMessage, error);
+    }
+  };
+
+  const onConfirmDisconnect = (evt?: MouseEvent) => {
+    setSubmitStartTime(Date.now());
+
+    if (evt) {
+      // TODO: disabled ts-checking because the EventTarget type is missing
+      // 'closest'. Figure out how to most nicely tidy this up, even if that
+      // entails trying again with managed components / react state.
+      const modalEl = evt.target.closest('#modal');
```

Comment:
`(evt.target as Element)`?

Reply:
Whoa, brilliant! That fixed the type error 🍻